app/controllers/comments_controller.rb

```diff
@@ -140,38 +140,40 @@ def moderator_create
   def update
     authorize @comment
 
-    if @comment.update(permitted_attributes(@comment).merge(edited_at: Time.zone.now))
-      Mention.create_all(@comment)
-
-      # The following sets variables used in the index view. We render the
-      # index view directly to avoid having to redirect.
-      #
-      # Redirects lead to a race condition where we redirect to a cached view
-      # after updating data and we don't bust the cache fast enough before
-      # hitting the view, therefore stale content ends up being served from
-      # cache.
-      #
-      # https://github.com/forem/forem/issues/10338#issuecomment-693401481
-      @on_comments_page = true
-      @root_comment = @comment
-      @commentable = @comment.commentable
-      @commentable_type = @comment.commentable_type
-
-      case @commentable_type
-      when "PodcastEpisode"
-        @user = @commentable&.podcast
-      when "Article"
-        # user could be a user or an organization
-        @user = @commentable&.user
-        @article = @commentable
+    Comment.transaction do
```

Comment:
I did double check that that [comments api](https://github.com/forem/forem/blob/main/app/controllers/api/v0/comments_controller.rb) provides index and show but not create or update (you can't comment or edit comments through the api - this controller should be the only place things are happening)

If we ever add management of comments through the api this behavior may want to be abstracted out of the controller action.

Reply:
The API is actually the first place I tried to do this. I didn't realize we did updated through the HTML endpoint at all, tbh. I thought it was only through the API.

> If we ever add management of comments through the api this behavior may want to be abstracted out of the controller action.

Do you know how we'd get the user performing the action outside the controller?

Reply:
oh - I meant via the api controller (in some hypothetical world where the CommentsController and the Api::CommentsController both supported update) - this was just agreeing with you - as close as I can tell this is the only place for a user to change a comment.